src/Grpc.AspNetCore.Server/Compression/GzipCompressionProvider.cs

```diff
@@ -0,0 +1,66 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System.IO;
+using System.IO.Compression;
+
+namespace Grpc.AspNetCore.Server.Compression
+{
+    /// <summary>
+    /// GZIP compression provider.
+    /// </summary>
+    public class GzipCompressionProvider : ICompressionProvider
```

Comment:
nit: API stability: once we declare the API as stable, we won't be able to add new methods to ICompressionProvider (and we might need to add some for e.g. pipepine-compatible compression support one day). Not a concern now, just FYI.

Reply:
I know. We'll do an API review at some point. ICompressionProvider might turn into an abstract base class so we can extend it.